### ISSUE TITLE
Non-ASCII filename raise exception when make singlehtml

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1508]
This error is reported by Mikami-san on Sphinx-Users (JP) ML
http://www.python.jp/pipermail/sphinx-users/2014-July/000997.html

```
# Sphinx version: 1.2.2
# Python version: 2.7.3
# Docutils version: 0.11 release
# Jinja2 version: 2.7.2
# Loaded extensions:
#   sphinx.ext.todo from C:\Python27\lib\site-packages\sphinx\ext\todo.pyc
#   wikibuilder from c:\Sphinx\cloud\wikibuilder.pyc
#   sphinx.ext.oldcmarkup from C:\Python27\lib\site-packages\sphinx\ext\oldcmarkup.pyc
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\sphinx\cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "C:\Python27\lib\site-packages\sphinx\application.py", line 212, in build
    self.builder.build_update()
  File "C:\Python27\lib\site-packages\sphinx\builders\__init__.py", line 209, in build_update
    self.build(['__all__'], to_build)
  File "C:\Python27\lib\site-packages\sphinx\builders\__init__.py", line 276, in build
    self.write(docnames, list(updated_docnames), method)
  File "C:\Python27\lib\site-packages\sphinx\builders\html.py", line 956, in write
    doctree = self.assemble_doctree()
  File "C:\Python27\lib\site-packages\sphinx\builders\html.py", line 916, in assemble_doctree
    tree = inline_all_toctrees(self, set(), master, tree, darkgreen)
  File "C:\Python27\lib\site-packages\sphinx\util\nodes.py", line 189, in inline_all_toctrees
    includefiles = map(str, toctreenode['includefiles'])
UnicodeEncodeError: 'ascii' codec can't encode characters in position 14-17: ordinal not in range(128)
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1508
- Originally reported by: Takayuki Shimizukawa
- Originally created at: 2014-07-12T14:29:15.244

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-13 01:43:20+00:00_

Reason 1: `sphinx.util.nodes.inline_all_toctrees` map filenames to `str` to combine into 1 file for singlehtml, latex, man and texinfo builders.

Reason 2: changes builder write unicode filename to changes html with 'latin1' encoding.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-07-13 01:46:30+00:00_
- Fix: Non-ASCII filename raise exception on make singlehtml, latex, man, texinfo and changes. Closes #1508

→ <<cset 269421bc5844ec6234b8c37bfb2a5dcbec609f27>>
